lib/python/rose/suite_engine_procs/cylc.py

```diff
@@ -130,19 +130,71 @@ def cmp_suite_conf(
         finally:
             os.unlink(new_suite_rc_processed)
 
-    def gcontrol(self, suite_name, host=None, engine_version=None, args=None):
-        """Launch control GUI for a suite_name running at a host."""
-        if not self.is_suite_registered(suite_name):
-            raise SuiteNotRegisteredError(suite_name)
-        if not host:
-            host = "localhost"
-        environ = dict(os.environ)
-        if engine_version:
-            environ.update({self.get_version_env_name(): engine_version})
-        fmt = r"nohup cylc gui --host=%s %s %s 1>%s 2>&1 &"
+    def gcontrol(self, suite_name, args=None):
+        """Launch control GUI for a suite_name."""
+        if 'DISPLAY' not in os.environ:
+            return
+        cylc_version = (
+            self.get_suite_contact(suite_name).get('CYLC_VERSION', ''))
+        self.popen.run_bg(
+            r'nohup',
+            r'bash',
+            r'-c',
+            (
+                r'exec env CYLC_VERSION=%s cylc gui %s %s' +
+                r' <"/dev/null" >"/dev/null" 2>&1'
+            ) % (cylc_version, suite_name, self.popen.list_to_shell_str(args)),
+            preexec_fn=os.setpgrp,
+            stdin=open(os.devnull),
+            stdout=open(os.devnull, "wb"),
+            stderr=open(os.devnull, "wb"),
+        )
+
+    def gscan(self, args=None):
+        """Launch suites scan GUI."""
+        if 'DISPLAY' not in os.environ:
+            return
         args_str = self.popen.list_to_shell_str(args)
-        self.popen(fmt % (host, suite_name, args_str, os.devnull),
-                   env=environ, shell=True)
+        self.popen.run_bg(
+            r'nohup',
+            r'bash',
+            r'-c',
+            r'exec cylc gscan %s <"/dev/null" >"/dev/null" 2>&1' % (args_str),
+            preexec_fn=os.setpgrp,
+            stdin=open(os.devnull),
+            stdout=open(os.devnull, "wb"),
+            stderr=open(os.devnull, "wb"),
+        )
+
+    def get_running_suites(self):
+        """Return a list containing the names of running suites."""
+        rootd = os.path.join(os.path.expanduser('~'), self.SUITE_DIR_REL_ROOT)
+        sub_names = ['.service', 'log', 'share', 'work', self.SUITE_CONF]
+        items = []
+        for dirpath, dnames, fnames in os.walk(rootd, followlinks=True):
+            if dirpath != rootd and any(
+                    name in dnames or name in fnames for name in sub_names):
```

Comment:
Is it perhaps more Pythonic to do ``any(name in dnames + fnames for ...)``?

Reply:
It is nicer syntax certainly. Not sure whether the extra `+` operation will have a penalty against the `or` short circuit logic. Not likely to have any impact here. I'll change the syntax just for the readability.

Reply:
I don't think `or` and `+` will do the same thing here?

```python
>>> a = [1,2,3]
>>> b = [4,5,6]
>>> [x for x in a or x in b]
[1, 2, 3]
>>> [x for x in a + b]
[1, 2, 3, 4, 5, 6]
```

Reply:
The original logic is more like:

```python
any(x in a or x in b for x in x_list)
# which should give the same results as:
any(x in a + b for x in x_list)
```